compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -471,16 +472,20 @@ fn collate_raw_dylibs(
         }
     }
 
-    // Rustc already signals an error if we have two imports with the same name but different
-    // calling conventions (or function signatures), so we don't have pay attention to those
-    // when ordering.
-    // FIXME: when we add support for ordinals, figure out if we need to do anything if we
-    // have two DllImport values with the same name but different ordinals.
+    fn compare_imports(x: &DllImport, y: &DllImport) -> Ordering {
```

Comment:
You can destructure to ensure that you matched all fields. Eg
```suggestion
    fn compare_imports(DllImport { x_name, x_ordinal, x_calling_convention, x_span }: &DllImport, DllImport { y_name, y_ordinal, y_calling_convention, y_span }: &DllImport) -> Ordering {
```
or
```suggestion
    fn compare_imports(x: &DllImport, y: &DllImport) -> Ordering {
        let DllImport { x_name, x_ordinal, x_calling_convention, x_span } = x;
        let DllImport { y_name, y_ordinal, y_calling_convention, y_span } = y;
```

Reply:
Made moot by petrochenkov's rewrite.